apps/api/src/app/workflows-v2/usecases/process-workflow-issues/process-workflow-issues.usecase.ts

```diff
@@ -123,28 +123,66 @@ export class ValidateAndPersistWorkflowIssuesUsecase {
     }
   }
   private async validateWorkflow(
-    command: ValidateWorkflowCommand
+    command: ProcessWorkflowIssuesCommand
   ): Promise<Record<keyof WorkflowResponseDto, RuntimeIssue[]>> {
     // @ts-ignore
     const issues: Record<keyof WorkflowResponseDto, RuntimeIssue[]> = {};
     await this.addTriggerIdentifierNotUniqueIfApplicable(command, issues);
     this.addNameMissingIfApplicable(command, issues);
     this.addDescriptionTooLongIfApplicable(command, issues);
+    this.processTags(command, issues);
 
     return issues;
   }
 
   private addNameMissingIfApplicable(
-    command: ValidateWorkflowCommand,
+    command: ProcessWorkflowIssuesCommand,
     issues: Record<keyof WorkflowResponseDto, RuntimeIssue[]>
   ) {
     if (!command.workflow.name || command.workflow.name.trim() === '') {
       // eslint-disable-next-line no-param-reassign
       issues.name = [{ issueType: WorkflowIssueTypeEnum.MISSING_VALUE, message: 'Name is missing' }];
     }
   }
+
+  private processTags(
+    command: ProcessWorkflowIssuesCommand,
+    issues: Record<keyof WorkflowResponseDto, RuntimeIssue[]>
+  ) {
+    const tags = command.workflow.tags?.map((tag) => tag.trim());
+
+    if (!tags.length) {
+      return;
+    }
+
+    const tagsIssues: RuntimeIssue[] = [];
+
+    const hasDuplications = tags.length !== new Set(tags).size;
+    if (hasDuplications) {
+      tagsIssues.push({ issueType: WorkflowIssueTypeEnum.DUPLICATED_VALUE, message: 'Duplicated tags' });
```

Comment:
i wonder if we should add the duplication values here or is the generic message will be enough.

Reply:
never mind i added those